tools/python/src/face_recognition.cpp

```diff
@@ -127,6 +136,57 @@ class face_recognition_model_v1
         return face_descriptors;
     }
 
+    std::vector<matrix<double,0,1>> batch_compute_face_descriptors_from_aligned_images (
+        const std::vector<numpy_image<rgb_pixel>>& batch_imgs,        
+        const int num_jitters
+    )
+    {
+        dlib::array<matrix<rgb_pixel>> face_chips;           
+        for (auto& img : batch_imgs) {
+
+            matrix<rgb_pixel> image;
+            if (is_image<unsigned char>(img))
+                assign_image(image, numpy_image<unsigned char>(img));
+            else if (is_image<rgb_pixel>(img))
+                assign_image(image, numpy_image<rgb_pixel>(img));
+            else
+                throw dlib::error("Unsupported image type, must be 8bit gray or RGB image.");
+
+            // Check for the size of the image
+            if ((image.nr() != 150) || (image.nc() != 150)) {
+                throw dlib::error("Unsupported image size, it should be of size 150x150.");
+            }
+
+            face_chips.push_back(image);        
+        }       
+
+        std::vector<matrix<double,0,1>> face_descriptors;
+        if (num_jitters <= 1)
+        {
+            // extract descriptors and convert from float vectors to double vectors
+            auto descriptors = net(face_chips, 16);            
+
+            auto next = std::begin(descriptors);
+            for (int i = 0; i < face_chips.size(); ++i)
+            {                   
+                face_descriptors.push_back(matrix_cast<double>(*next++));                
+            }
+            DLIB_ASSERT(next == std::end(descriptors));            
+        }
+        else
+        {
+            // extract descriptors and convert from float vectors to double vectors
+            auto fimg = std::begin(face_chips);
+            for (int i = 0; i < face_chips.size(); ++i)
```

Comment:
This loop doesn't use `i`.  Use a range loop over `face_chips` instead like `for (auto& fimg : face_chips) ...`.